indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -577,7 +587,9 @@ public Response getCompleteTasks(
             DateTimes.EPOCH,
             status.getStatusCode(),
             status.getDuration(),
-            TaskLocation.unknown())
+            TaskLocation.unknown(),
+            taskStorageQueryAdapter.getTask(status.getId()).get().getDataSource()
```

Comment:
maybe add a method to TaskStorageQueryAdapter getCreatedDateTimeAndDataSource which fetches datasource and created time in single query instead of issuing multiple queries to metadata storage.

Reply:
Added getCreatedDateTimeAndDataSource in TaskStorageQueryAdapter